daemon/remote/cmd/cmuxd-remote/agent_launch.go

```
@@ -323,10 +323,14 @@ func writeShimIfChanged(path string, content string) error {
 		tempFile.Close()
 		return err
 	}
-	if err := tempFile.Close(); err != nil {
+	// Set the final executable mode while the descriptor is still open. This
+	// prevents a close-to-chmod window in which another same-UID process could
+	// observe or replace the temporary file.
+	if err := tempFile.Chmod(0755); err != nil {
+		tempFile.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `tempFile.Close` is not checked

(errcheck)